src/Components/WebAssembly/Server/src/AuthenticationStateSerializer.cs

```diff
@@ -17,6 +17,14 @@ internal sealed class AuthenticationStateSerializer : IHostEnvironmentAuthentica
     private readonly PersistingComponentStateSubscription _subscription;
 
     private Task<AuthenticationState>? _authenticationStateTask;
+    private AuthenticationStateData? _authenticationStateData;
+
+    [PersistentComponentStateStore]
+    public AuthenticationStateData? AuthStateData
+    {
+        get => _authenticationStateData;
+        set => _authenticationStateData = value;
+    }
 
     public AuthenticationStateSerializer(PersistentComponentState persistentComponentState, IOptions<AuthenticationStateSerializationOptions> options)
```

Comment:
We don't have to manually RegisterOnPersisting when we use `[SupplyParameterFromPersistentComponentState]`. That makes `_subscription` field and `persistentComponentState` param redundant.